trio/_socket.py

```diff
@@ -472,44 +478,45 @@ def __getattr__(self, name):
             return getattr(self._sock, name)
         raise AttributeError(name)
 
-    def __dir__(self):
-        return super().__dir__() + list(self._forward)
+    def __dir__(self) -> Iterable[str]:
+        yield from super().__dir__()
+        yield from self._forward
 
-    def __enter__(self):
+    def __enter__(self) -> Self:
         return self
 
-    def __exit__(self, *exc_info):
+    def __exit__(self, *exc_info: object) -> None:
```

Comment:
why not fix this signature instead of just hacking onto the hack?

Reply:
Ah right, it should have the normal trio of exception values. Will do